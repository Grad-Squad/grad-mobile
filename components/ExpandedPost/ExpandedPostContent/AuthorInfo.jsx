import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Image, TouchableOpacity } from 'react-native';
import EduText from 'common/EduText';
import { Colors } from 'styles';
import { AssetsConstants } from 'constants';
import ScreenNames from 'navigation/ScreenNames';
import { useNavigation } from '@react-navigation/native';

const AuthorInfo = ({ profilePicture, name, profileId }) => {
  const navigation = useNavigation();
  return (
    <View style={styles.profileInfoContainer}>
      <TouchableOpacity
        onPress={() => navigation.navigate(ScreenNames.PROFILE, { profileId })}
      >
        <Image
          style={styles.profileImage}
          source={
            profilePicture
              ? {
                  uri: profilePicture,
                }
              : AssetsConstants.images.defaultProfile
          }
        />
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() => navigation.navigate(ScreenNames.PROFILE, { profileId })}
      >
        <EduText style={styles.authorName}>{name}</EduText>
      </TouchableOpacity>
    </View>
  );
};

AuthorInfo.propTypes = {
  profilePicture: PropTypes.string,
  name: PropTypes.string.isRequired,
  profileId: PropTypes.number.isRequired,
};
AuthorInfo.defaultProps = { profilePicture: undefined };

export default React.memo(AuthorInfo);

const styles = StyleSheet.create({
  profileInfoContainer: {
    alignItems: 'center',
    marginTop: 'auto',

    flexBasis: '20%',

    marginRight: 5,
  },
  profileImage: {
    borderRadius: 50,
    width: 50,
    height: 50,
    // borderWidth: 1,
    // borderColor: Colors.black,
  },
  authorName: {
    fontSize: 10,
    textAlign: 'center',
    marginTop: 5,
  },
});
