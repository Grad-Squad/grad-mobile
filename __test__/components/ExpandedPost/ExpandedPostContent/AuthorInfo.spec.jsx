import React from 'react';
import { render } from '@testing-library/react-native';
import AuthorInfo from 'components/ExpandedPost/ExpandedPostContent/AuthorInfo';

it.each([{ uri: 'https://picsum.photos/200/300' }, null])(
  'should match snapshot when profilePicture=%s',
  (profilePicture) => {
    const authorInfo = render(
      <AuthorInfo profilePicture={profilePicture} name="name" profileId={1} />
    );
    expect(authorInfo.toJSON()).toMatchSnapshot();
  }
);
