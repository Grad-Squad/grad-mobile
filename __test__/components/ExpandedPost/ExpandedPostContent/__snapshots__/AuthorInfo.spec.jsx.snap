// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when profilePicture=https://picsum.photos/200/300 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexBasis": "20%",
      "marginRight": 5,
      "marginTop": "auto",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://picsum.photos/200/300",
        }
      }
      style={
        Object {
          "borderColor": "#000",
          "borderRadius": 50,
          "borderWidth": 0.1,
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Lato_400Regular",
          },
          Object {
            "color": "#3C3C3C",
          },
          Object {
            "fontSize": 10,
            "marginTop": 5,
            "textAlign": "center",
          },
        ]
      }
    >
      name
    </Text>
  </View>
</View>
`;

exports[`should match snapshot when profilePicture=undefined 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexBasis": "20%",
      "marginRight": 5,
      "marginTop": "auto",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderColor": "#000",
          "borderRadius": 50,
          "borderWidth": 0.1,
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Lato_400Regular",
          },
          Object {
            "color": "#3C3C3C",
          },
          Object {
            "fontSize": 10,
            "marginTop": 5,
            "textAlign": "center",
          },
        ]
      }
    >
      name
    </Text>
  </View>
</View>
`;
